review_pr.py

```
@@ -18,37 +32,55 @@
+        response.raise_for_status()
+        return response.json().get("head", {}).get("ref")
+    except requests.RequestException as e:
+        logger.error(f"Error fetching PR branch: {e}")
+        return None
 
- # Fetch modified files
-def get_modified_files(pr_number):   
-    url = f"{GITHUB_API_BASE_URL}/repos/{REPO_NAME}/pulls/{pr_number}/files"
-    response = requests.get(url, headers=GITHUB_HEADERS)
-    return response.json() if response.status_code == 200 else []
+# It fetches the modified files in the PR using github api
+def get_modified_files(pr_number):
+    try:
+        url = f"{GITHUB_API_BASE_URL}/repos/{REPO_NAME}/pulls/{pr_number}/files"
+        response = requests.get(url, headers=GITHUB_HEADERS)
+        response.raise_for_status()
+        return response.json()
+    except requests.RequestException as e:
+        logger.error(f"Error fetching modified files: {e}")
+        return []
 
-# Fetch file content from branch
+# It fetches the file content using github api anf this content we will pass to openai for review
 def get_file_content(file_path, branch_name):
-    url = f"{GITHUB_API_BASE_URL}/repos/{REPO_NAME}/contents/{file_path}?ref={branch_name}"
-    response = requests.get(url, headers=GITHUB_HEADERS)
-    if response.status_code == 200:
+    try:
+        url = f"{GITHUB_API_BASE_URL}/repos/{REPO_NAME}/contents/{file_path}?ref={branch_name}"
+        response = requests.get(url, headers=GITHUB_HEADERS)
+        response.raise_for_status()
         return base64.b64decode(response.json()["content"]).decode("utf-8")
-    return None
+    except requests.RequestException as e:
+        logger.error(f"Error fetching file content: {e}")
+        return None
 
-# Analyze code with OpenAI
+# It will review the code using openai api and return the response in json format 
+# we can change the prompt as per our requirement
+#eg if you dont want nested ifs and too many loops you can add that in prompt and review will done accordingly
```

Better error handling would involve failing silently by returning an empty response or a standard error response when data can't be obtained, and logging these incidents.

**Suggested Code:**
```py
return {'review': '[No review available]', 'comments': []}
```